cg/meta/store/base.py

```diff
@@ -29,26 +42,114 @@ def add_new_analysis(bundle_data, case_obj, status, version_obj):
 
 def reset_case_action(case_obj):
     """ Resets action on case """
+
     case_obj.action = None
 
 
-def get_files(deliverables: dict, pipeline: str) -> dict:
+def parse_files(deliverables: dict, pipeline: str) -> list:
     """Get all deliverable files from the pipeline"""
 
-    data = [
-        {"path": file["path"], "tags": get_tags(file, pipeline), "archive": False}
+    pipeline_tags = HK_TAGS[pipeline]
+    files = _parse_files_non_index(deliverables, pipeline_tags)
+    _check_mandatory_tags(files, PIPELINE_TAGS[pipeline])
+    _convert_tags(files, PIPELINE_TAGS[pipeline], "tags")
+    index_files = _parse_files_index(deliverables, pipeline_tags)
+    _convert_tags(index_files, PIPELINE_TAGS[pipeline], "index_tags")
+
+    files.extend(index_files)
+
+    return files
+
+
+def _parse_files_non_index(deliverables: dict, pipeline_tags: list) -> list:
+    """ Get all files that are not index files from the deliverables file """
+
+    return [
+        {"path": file["path"], "tags": get_tags(file, pipeline_tags), "archive": False}
         for file in deliverables["files"]
     ]
 
-    return data
+
+def _parse_files_index(deliverables: dict, pipeline_tags: list) -> list:
+    """ Get all index files from the deliverables file """
+
+    return [
+        {"path": file["path_index"], "tags": get_tags(file, pipeline_tags), "archive": False}
+        for file in deliverables["files"]
+        if file["path_index"]
+    ]
 
 
-def get_tags(file: dict, pipeline: str) -> list:
+def get_tags(file: dict, pipeline_tags: list) -> list:
     """Get all tags for a file"""
 
-    all_tags = [file["format"], file["id"], file["step"], file["tag"], pipeline]
+    all_tags = [file["format"], file["id"], file["step"], file["tag"]]
+    all_tags.extend(pipeline_tags)
     unique_tags = set(all_tags)
     only_existing_tags = unique_tags - set([None])
     sorted_tags = sorted(list(only_existing_tags))
 
     return sorted_tags
+
+
+def build_bundle(config_data: dict, analysisinfo_data: dict, deliverables: dict) -> dict:
+    """Create a new bundle to store in Housekeeper"""
+
+    analysis_type = config_data["samples"][0]["type"]
+
+    data = {
+        "name": config_data["case"],
+        "created": analysisinfo_data["date"],
+        "pipeline_version": analysisinfo_data["version"],
+        "files": parse_files(deliverables, analysis_type),
```

Comment:
It is just that it is not immediately obvious that the parse_files def will return `files` as it was when it was called `get_files`. Usually one would expect that a parse will check, get and set stuff within the parse function but perhaps not return more than a boolean for successful parsing or not. Lifting it from the dict will give you more lines to be more explicit if you want. On the other hand this is recorded in the docstring and annotations so I think you are fine.

Reply:
So should i change it or can we resolve this?

Reply:
It would be great to close this one sooner rather than later. I perfectly fine with this being resolved. Great work